### DISCUSSION TITLE
How can I get Prometheus monitoring metrics of all REST endpoints?

### DISCUSSION THREAD

**Discussion by jopime** [qdrant/qdrant#4248] [Q&A]
when I log in I can only see a small set of the endpoints, it looks like the ones defined in this [whitelist](https://github.com/qdrant/qdrant/blob/34f7f8ec2d8455c0b3fe0f1fb134907ba7816a08/src/common/metrics.rs#L20).
Is there any way to see the metrics of other endpoints? such as alias creations, deletion, etc.

Is there any plan to support this? or it will not be supported in future

**Comment by timvisee** [comment]
You're correct, only the endpoints listed in the whitelist are shown.

We currently have no plans to expose other endpoints too.

Our reasoning for that is that stats for the other endpoints are not really that valuable. At the same time they bloat the `/metrics` output. That's why we limited it to endpoints for which we think it makes sense.

**Comment by jopime** [comment]
@timvisee Thanks for your answer, I understand the reason for the current decision.

But I will explain our case to you, in the deployment process we are suffering failures related to the management of aliases, switching collections, etc.

By not being able to obtain metrics from these endpoints, we are losing the option to monitor and alarm these events.

Would it be possible for this whitelist to be configurable?
  Do I add it as issue or request?

Thank you

**Reply by timvisee** [reply]
The only thing it will expose is how many times those endpoints are called, not with what data. Are you sure that would be enough?

I'd think that triggering warnings at a different stage might make more sense; possibly at the point where clients make such calls, because then you'd have all the additional context you might need.

In other words, metrics will not tell you what, when and how aliases are switched.

To add, you might find valuable information on this in the Qdrant logs.

**Comment by jopime** [comment]
Yes, I am aware of it.
 
If we could collect the number of times these endpoints are called along with the status obtained, we could create a preventive alarm policy, which would allow us to anticipate bad behavior from Qdrant and be able to take action.
 
Be aware of when your performance is starting to degrade and be able to later do a more in-depth review of the reason, now using all the logs and the full context. 
 
To give you a little more context, when deploying and provisioning data to Qdrant, we are suffering 500 timeout errors, mainly related to calls to collections and aliases endpoints.
Not having this data exposed in Prometheus complicates how to create graphana alarms and the rest of the monitoring.
 
Since the problems are not related to searching for points in the database, the current metrics do not alert us to bad behavior until it is too late and Qdrant has already crashed.

Could I then create an issue or change request?
Thanks in advance.
